lib/global-styles.php

```diff
@@ -298,7 +298,7 @@ function gutenberg_experimental_global_styles_get_block_data() {
 	$block_data = array(
 		'global' => array(
 			'selector' => ':root',
-			'supports' => array( 'background-color' ),
+			'supports' => array( '--wp--style--color--link', 'background-color', 'font-size' ),
```

Comment:
Can you clarify what impact a change here has?

Reply:
The supports array of each context is used in two places:

- The global styles sidebar in edit-site https://github.com/WordPress/gutenberg/pull/24250 to decide whether a UI control should be shown. For example, this PR adds the link color & font size controls in the UI for the global context.

- As an allow filter when processing the corresponding json for merging the different origins (core, theme, user). For example, this PR, tells the global styles resolver to take into account the font size, link color, and background color of the `lib/experimental-theme.json` file that comes with core --- other properties will be ignored.

This is the only context for which we take the data (selector, supports) from a source other than the block.json. If we end up modelling this as properties of a site block they could be embedded in its supports declarations.

Added testing instructions in the issue description for clarity!

Reply:
> The global styles sidebar in edit-site #24250 to decide whether a UI control should be shown. For example, this PR adds the link color & font size controls in the UI for the global context.

I guess in my thinking the UI should be defined in theme.json as themes could decide to show some things and hide others but it's something we can discuss later and iterate on as needed.

> As an allow filter when processing the corresponding json for merging the different origins (core, theme, user). For example, when this PR lands, the global styles resolver would take into account the font size, background color, and link color when processing lib/experimental-theme.json file that comes with core --- other properties will be ignored.

Why would it take into consideration just this list and not all what's defined on theme.json?

Reply:
In other words, why it's different from this list https://github.com/WordPress/gutenberg/pull/24296/files#diff-4339a9e86722fd029c231f3fb65f0ad2R271 ?

Reply:
> I guess in my thinking the UI should be defined in theme.json as themes could decide to show some things and hide others but it's something we can discuss later and iterate on as needed.

:+1: 

For the other comment: I've pushed a change for this to be declared as if it was a block.json. It has some issues but we can discuss over here https://github.com/WordPress/gutenberg/pull/24298/files#r464115543